### DISCUSSION TITLE
Plans To Update To Tensorrt 10.x?

### DISCUSSION THREAD

**Discussion by barrycarey** [SthPhoenix/InsightFace-REST#133] [General]
Hello!

The project using 8.6.1 makes developing locally pretty painful since pretty much everything Nvidia needs to be downgraded. 

I've started looking at what it will take to upgrade, however, I'm new to Tensorrt so it has been slow going so far.

**Comment by SthPhoenix** [comment]
Done. 
Though I still left `nvcr.io/nvidia/tensorrt:23.05-py3` as base image since I haven't finished testing yet. If you want to try TRT10 in container - just switch base image to `nvcr.io/nvidia/tensorrt:24.05-py3` in `src/Dockerfile_trt`

**Reply by barrycarey** [reply]
I'll give it a shot tonight!  

However, there were some depreciated methods, set_max_woekspace_size as an example, that I believe would require a code change on your end.

**Reply by SthPhoenix** [reply]
It's already changed ) It should work now with both TRT10 and TRT8.6

**Reply by barrycarey** [reply]
Perfect, thanks!

Are you open to pull requests?

I'm getting ready to generate embeddings for 2 billion images and there are some changes I was going to make in a fork for my work flow.  Such as a flow to process single images, bubbling up HTTPExceptions, and exposing retry settings to configuration. 

If you think it would be useful I can send them back.

**Reply by SthPhoenix** [reply]
Sure, if it won't broke anything else - why not.

**Reply by barrycarey** [reply]
Did r100-arcface-msfdrop75 and r50-arcface-msfdrop75 not get moved over to models.json for any reason or was it an oversight?

**Reply by SthPhoenix** [reply]
Those are some old InsightFace models, they are still somewhere in original repo, but yes, now they are pretty useless.

**Reply by barrycarey** [reply]
I was originally testing with Compreface using r100-arcface-msfdrop75 since they listed it as the most accurate.   Figured I'd continue with that here to stay consistent. 

Is there a specific recognition model you would recommend?  I'm interested in accuracy over performance.

**Comment by SthPhoenix** [comment]
For now best option IMO is `adaface_ir101_webface12m`
